### PR TITLE
Mj 4356 test migration

### PR DESCRIPTION
**Pull request by micahjones13**
## Brief Description

Replaces Cypress with Playwright. 
- Uninstalls cypress
- deletes the **.e2e.js files in each test dir of each component
- adds new tests/ dir in web-components
- adds playwright config 
  - Currently tests on chromium browsers
  - is configured to reuse the stencil server at localhost:3333
- All e2e tests have been migrated to be in Playwright
- Updated github actions to run playwright tests
    - GA tests are running ~1 min faster and have less flake (flake is still there for first test, but playwright will re-run it by 
    default so we don't have to)

This doesn't need a changeset since it's a test change only. 
## JIRA Link

https://rocketcom.atlassian.net/browse/ASTRO-4356

## General Notes

Tests were migrated from cypress -> playwright in a 1:1 (for the most part). While doing this, Frances and I wrote recommended additions to our tests in comments to be added in the future. 
I will also drop the about 70 unnecessary commit messages when I squash/merge this.

## Motivation and Context

Cypress was too slow and much too flaky, especially in GA. Playwright is faster and less flaky. 

## Types of changes

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change

## Checklist

- [ ] This PR adds or removes a Storybook story.
- [ ] I have added tests to cover my changes.
- [ ] Regressions are passing and/or failures are documented
- [ ] Changes have been checked in evergreen browsers

### CODE REVIEW

**Review by markacianfrani — APPROVED** [review]
can we update the CONTRIBUTING.md ?

**Review by brikiernan — APPROVED** [review]
Nice job @micahjones13!

**Review by kiley-mitti — APPROVED** [review]
Oh man you guys did so much work here. Well done!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -84,35 +82,27 @@ All new components should have an associated Storybook story that displays any v
...
+- `npm run test.e2e` - Runs all e2e tests in Chromium, Firefox, and WebKit.
+- `npm run test.cr` - Runs e2e tests in Chromium.
+- `npm run test.ff` - Runs e2e tests in Firefox.
+- `npm run test.wk` - Runs e2e tests in WebKit.
+- `npm run test` - Runs all unit and e2e tests.
 
 ## Submitting your first PR
 
-Astro uses [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) for writing git messages.
+Astro loosely follows [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) for writing git messages.
```
Comment (markacianfrani) [inline CONTRIBUTING.md:105]: nice

**Pull request merged**